src/api.js

```diff
@@ -171,33 +171,32 @@ export default class BounceClient {
 
     /**
      * Fetch information about a club
-     * @param {String} name
+     * @param {String} id
      */
-    async getClub(name) {
-        return await this._request('GET', '/clubs/' + name);
+    async getClub(id) {
+        return await this._request('GET', '/clubs/' + id);
     }
 
     /**
      * Returns a list of clubs that match the given query.
      * @param {String} query
      */
-    async searchClubs(query) {
-        if (!query) return await this._request('GET', '/clubs/search');
-        return await this._request('GET', `/clubs/search?query=${query}`);
+    async searchClubs(params) {
+        return await this._request('GET', '/clubs/search', null, params);
     }
 
     /**
      * Create a new club with the given properties
-     * @param {String} name
+     * @param {String} id
      * @param {String} description
      * @param {String} websiteUrl
      * @param {String} facebookUrl
      * @param {String} instagramUrl
      * @param {String} twitterUrl
      */
-    async createClub(name, description, websiteUrl, facebookUrl, instagramUrl, twitterUrl) {
+    async createClub(id, description, websiteUrl, facebookUrl, instagramUrl, twitterUrl) {
```

Comment:
I don't think you should change this. The user should still provide the club name on create and I think the id is auto-generated on the server side.